content/general/concepts/algorithm/terms/merge-sort/merge-sort.md

```diff
@@ -0,0 +1,141 @@
+---
+Title: 'Merge Sort Algorithm' 
+
+Description: 'Returns a sorted array using divide and conquer.' 
+
+Subjects: 
+  - 'Computer Science'
+  - 'Interview Prep'
+  - 'Code Foundations'
+Tags:
+  - 'Sorting Algorithms'
+  - 'Algorithms'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+**Merge sort** is a divide-and-conquer sorting [algorithm](https://www.codecademy.com/resources/docs/general/algorithm) that breaks down an [array](https://www.codecademy.com/resources/docs/java/arrays) into smaller arrays, sorts them, and then combines the subarrays back together to return a sorted array. 
+
+
+
+## MergeSort Method
+
+Divide:
+  - Base Case: If the input array `data` has only one element, it's already sorted, so return.
+  - Split: Divide the array into two halves, `left` and `right`, of approximately equal size.
+
+Conquer:
+  
+  - [Recursive](https://www.codecademy.com/learn/java-algorithms/modules/recursion-apcs/cheatsheet) Calls: Repeatedly call `mergeSort` on each half, `(left)` and `(right)`, to create subarrays so they are sorted independently.
+
+Combine:
+  
+  - Merge: Call the `merge` function to merge the two sorted halves (each initially of length 1) `left` and `right`, and combine them back into the original array `data` in a sorted manner.
+
+
+### Implementation Pt.1 
+
+```java
+public static void mergeSort(int[] data){
+    if(data.length == 1){
+        return;
+    }
+    int n1 = data.length/2;
+    int n2 = data.length - n1;
+    int[] left, right;
+    //make left and right subarrays
+    left = new int[n1];
+    right = new int[n2];
+    System.arraycopy(data, 0, left, 0, n1);
+    System.arraycopy(data, n1, right, 0, n2);
+    //recursively split left and right arrays
+    mergeSort(left);
+    mergeSort(right);
+    //once array is @ length 1, the left & right array will be merged
+    merge(data, left, right);
+}
+
+```
+## 
+
+### Merge Method
+
+1. Initialize: 
+   - Set indices `l`, `r`, and `k` to 0 to track the positions in `left`, `right`, and `data`.
+2. Compare and Merge:
+   - While both `left` and `right` have elements remaining:
+     - Compare the elements at `left[l]` and `right[r]`.
+     - Copy the smaller element into `data[k]` and increment the corresponding index (`l` or `r`).
+     - Increment `k` to move to the next position in `data`.
+3. Copy Remaining Elements:
+   - If any elements remain in `left`, copy them directly into the remaining positions in `data`.
+   - Similarly, if any elements remain in `right`, copy them into the remaining positions in `data`.
+
+## Implementation Pt.2
+
```

Comment:
again give a basic desc here for the code snippet down below @JustinhSE

Reply:
okay, just added it @dakshdeepHERE